pennylane/templates/utils.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2018 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""
+Utility functions used in the templates.
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from collections.abc import Iterable
+import numpy as np
+from pennylane.qnode import Variable
+
+
+def _check_no_variable(arg, arg_str, msg=None):
+    if msg is None:
+        msg = "The argument {} can not be passed as a QNode parameter.".format(arg_str)
+    for a, s in zip(arg, arg_str):
+        if isinstance(a, Variable):
+            raise ValueError(msg)
+        if isinstance(a, Iterable):
+            if any([isinstance(a_, Variable) for a_ in a]):
+                raise ValueError(msg)
+
+
+def _check_wires(wires):
+    """Standard checks for the wires argument.
+
+    Args:
+        wires (int or list): (subset of) wires of a quantum node
+
+    Return:
+        list: wires as a list, int: number of wires
+    """
+    if isinstance(wires, int):
+        wires = [wires]
+
+    msg = "Wires must be a positive integer or a " \
+           "list of positive integers; got {}.".format(wires)
+    if not isinstance(wires, Iterable):
+        raise ValueError(msg)
+    if not all([isinstance(w, int) for w in wires]):
+        raise ValueError(msg)
+    if not all([w >= 0 for w in wires]):
+        raise ValueError(msg)
+    return wires, len(wires)
```

Comment:
Maybe it might be a bit unexpected to have ``len(wires)`` returned too as the second object here. Would there be any downside to operate with ``len(wires)``, once ``_check_wires(wires)`` returned ``wires``?

Reply:
Sure, but since it is an internal function users don't have to access it. Ok to leave for now? We can easily change this!